### PR TITLE
feat(plaud): add Google/Apple connect paths via paste-token + Connector extension (fixes #65)

### LINKED ISSUE #65: [Feature]: Add Google/Apple OAuth sign-in to support users who registered via those methods
### Problem / Use Case

Hi, I think I may found a limitation in the app. I signed up for Plaud using Google sign-in. When I connect my account in OpenPlaud via the email OTP flow, the sync completes successfully but returns zero recordings and zero devices, even though I have 4 recordings visible on web.plaud.ai.

I grabbed the bearer token from web.plaud.ai (logged in via Google) and compared the JWT payloads. The two tokens have different user IDs, suggesting the OTP login accesses a different account than the Google-linked one ?

Manually injecting the web token into the database fixed the issue,sy nc immediately found all 4 recordings and my device.

Could you please add the option to sign in with Google/Apple? Thanks!

### Proposed Solution

Add Google and Apple sign-in as authentication options alongside the existing email OTP flow, so users connect with the same method they used to create their Plaud account.

### Alternatives Considered

_No response_

### Feature Category

Authentication

### Priority

Important

### Mockups / Examples

_No response_

### Additional Context

_No response_

### Checklist

- [x] I have searched existing issues and feature requests to avoid duplicates
- [x] This feature aligns with OpenPlaud's goals (self-hosted, privacy-focused)
- [ ] I am willing to contribute to implementing this feature

### PR DESCRIPTION
**Pull request by kacperkwapisz**
## Description

Adds end-to-end support for connecting Plaud accounts that were created via Google or Apple sign-in.

Plaud's email-OTP login signs users into a different account than Plaud's Google/Apple sign-in flow — even when both share the same email address. The OTP flow looks like it succeeds but sync returns zero recordings because the access token belongs to an empty shadow account, while the user's real recordings live under their SSO identity ([#65](https://github.com/openplaud/openplaud/issues/65)).

Real Sign in with Google / Apple from OpenPlaud's origin is structurally blocked by Google's authorized-origins policy on Plaud's OAuth client (`346186524912-…apps.googleusercontent.com`), confirmed by HAR capture of the upstream flow. This PR ships the only honest workaround that works today, on two tiers:

1. **Easy path** — a companion [openplaud/connector](https://github.com/openplaud/connector) browser extension (separate AGPL-3.0 repo, scaffolded and pushed in the same session) that bridges a logged-in `web.plaud.ai` session to the OpenPlaud connect screen. No copy-pasting, no devtools.
2. **Manual fallback** — paste an access token grabbed from a logged-in `web.plaud.ai` session with documented devtools steps. Same encryption + validation path as the OTP flow.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Code refactoring
- [x] Documentation update

## Related Issues

Fixes #65

## Changes Made

- New `POST /api/plaud/auth/connect-token` route. Validates JWT shape + `exp`, SSRF-guards `apiBase`, derives `plaudEmail` via best-effort `/user/me`, hands off to the shared persistence helper.
- Extract `persistPlaudConnection` from the verify route into `src/lib/plaud/persist-connection.ts`. OTP flow and paste-token flow now go through identical workspace discovery, end-to-end `/device/list` validation, AES-256-GCM encryption, and per-user-scoped upsert. OTP behaviour unchanged.
- Add `decodeAccessTokenExpiry` (UX-only JWT payload parse, no signature check) and `fetchPlaudUserMeEmail` (best-effort, swallows failures) helpers in `src/lib/plaud/auth.ts`.
- New shared `<PlaudConnectTabs />` component used by both the onboarding dialog and the standalone onboarding page. Three tabs:
  - **Sign in with Plaud** (default) — feature-detects `window.__openplaudConnector` (injected by the connector extension), shows install CTA otherwise.
  - **Email code** — existing OTP flow.
  - **Paste token** — manual fallback with step-by-step token-grab instructions and a region selector.
- Connector detection re-polls for ~10s after mount so installing the extension while the page is open Just Works.
- README gets a new "Signed up to Plaud with Google or Apple?" subsection covering both paths.
- CHANGELOG entry under `[Unreleased] → Added`.

No DB schema changes. No env var changes. No new dependencies.

## Screenshots (if applicable)

N/A — UI changes are tabbed-form additions matching the existing dialog/onboarding visual language. Verified locally via `pnpm dev`.

### Test Environment

- OS: macOS
- Node version: per `package.json` engines (Bun for migrations, pnpm 10 for everything else)
- Browser: Chrome (for connector extension verification)

### Test Steps

```bash
pnpm format-and-lint   # 0 issues, 158 files checked
pnpm type-check        # 0 errors
pnpm test              # 64 passing, 3 skipped, 0 failing
```

New regression test `src/tests/regressions/65-paste-token.test.ts` (12 tests) covers:

- `decodeAccessTokenExpiry`: valid JWT → Date, malformed input → null, missing/non-numeric `exp` → null, base64url padding edge cases.
- Route validation: 401 without session, 400 on missing/malformed/expired token, 400 on non-`plaud.ai` `apiBase` (SSRF guard), `Bearer ` prefix stripped before forwarding to Plaud.
- Happy path: validates against `/device/list`, persists encrypted token (asserts the stored value is *not* the raw JWT), captures lowercased email from `/user/me`, upserts the device row scoped to the session user.

Existing regression suite (`66-eu-otp-permissions.test.ts`) still passes — verifies the OTP path's WT/UT behaviour is unchanged after the `persistPlaudConnection` refactor.

## Checklist

- [x] My code follows the style guidelines of this project (`pnpm format-and-lint`)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`pnpm test`)
- [x] Type checking passes (`pnpm type-check`)
- [x] Any dependent changes have been merged and published

## Database Changes

- [x] No schema changes in this PR.

## Breaking Changes

None. The OTP flow continues to work for users whose Plaud account was created with email/password. The new connect screen defaults to "Sign in with Plaud" but degrades gracefully — when the connector extension isn't installed it surfaces the install CTA plus inline links to the email-code and paste-token tabs.

## Additional Notes

The connector extension lives in a sibling repo so it can ship on its own cadence (Chrome Web Store / Firefox AMO reviews are independent of OpenPlaud releases) and keep this repo's bundle/lint config clean. Initial scaffold commit at [openplaud/connector@b0abdab](https://github.com/openplaud/connector/commit/b0abdab). Outstanding before extension launch: real PNG icons (TODO in manifest), Chrome Web Store / AMO submission. The README install link currently points to the GitHub repo; once published it should swap to the store listing.

Architecture details for the bridge live in the connector repo's README. Token never crosses external infrastructure: `web.plaud.ai` → service worker → OpenPlaud tab → OpenPlaud server, all over local IPC except the final HTTPS POST to a host the user controls.

## For Reviewers

- **Security**: the new `/api/plaud/auth/connect-token` route reuses the exact same SSRF guard (`isValidPlaudApiUrl`), encryption (AES-256-GCM via `lib/encryption`), and userId-scoped UPSERT discipline as the OTP verify route. The `decodeAccessTokenExpiry` JWT decode is explicitly UX-only — Plaud is the verifier on the `/device/list` round-trip below it.
- **Refactor blast radius**: `verify/route.ts` shrunk by ~115 lines after extracting `persistPlaudConnection`. The `66-eu-otp-permissions.test.ts` regression test pins the WT/UT behaviour and still passes; that's the load-bearing assertion that the OTP path is unchanged.
- **UI**: the third tab (paste-token) is a real fallback, not just decorative. Walk through the disclosure copy and the region-selector UX — the goal was to make the manual path tractable for non-technical users without screen-recording a video.

---
## Summary by cubic
Adds support for connecting Plaud accounts created with Google or Apple via a paste-token flow and a browser-extension bridge, plus a unified connect UI. Improves error handling and makes connection writes atomic for reliability. Fixes #65 where the OTP flow linked a shadow account and returned no recordings.

- **New Features**
  - `POST /api/plaud/auth/connect-token`: accepts a Plaud access token and region (`apiBase`), validates via `/device/list`, encrypts (AES-256-GCM), and upserts the connection; rejects non-JWT/expired tokens and non-`plaud.ai` API bases; best-effort email enrichment via `/user/me`.
  - `<PlaudConnectTabs />`: shared connect UI with three methods — “Sign in with Plaud” (via `openplaud/connector`), “Email code”, and “Paste token” with region selector and instructions; detects the connector with re-polling for ~10s; README and CHANGELOG updated.
  - Shared `persistPlaudConnection`: both OTP (`POST /api/plaud/auth/verify`) and paste-token flows use the same workspace discovery, end-to-end validation, encryption, and device upsert.

- **Bug Fixes**
  - Correct error classification: new `isUserActionablePlaudError` treats bare `Plaud API error: ...`, Plaud 4xx, and “Invalid API base” as 400; Plaud 5xx return 500; `persistPlaudConnection` re-throws original errors and both routes use the helper.
  - Prevent duplicate connections: wrap upsert + device reconciliation in a single `db.transaction()` guarded by a per-user `pg_advisory_xact_lock`.
  - Drop Plaud email from connect diagnostics to avoid PII in logs.
  - UI: replace escaped Unicode with literal characters in JSX.

<sup>Written for commit 640f264d199017f9435136172fc6b3368567f442. Summary will update on new commits.</sup>

### CODE REVIEW

File: src/lib/plaud/persist-connection.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    const client = new PlaudClient(accessToken, apiBase, resolvedWorkspaceId);
+    let deviceList: PlaudDeviceListResponse;
+    try {
+        deviceList = await client.listDevices();
+    } catch (err) {
+        console.warn(
+            "[plaud/persist] device list validation failed:",
+            err instanceof Error ? err.message : err,
+        );
+        throw new Error(
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/lib/plaud/persist-connection.ts]: Fixed in 7174477. Valid concern.

`persistPlaudConnection` no longer wraps the underlying error — re-throws verbatim so the HTTP status survives. Both route handlers now route through a new `isUserActionablePlaudError(msg)` helper in `src/lib/plaud/auth.ts` that:

- matches `Plaud API error (4xx): …` → 400 (auth/permission/bad-input — user's responsibility)
- matches `Invalid API base` → 400 (our SSRF guard)
- everything else (5xx, bare `Plaud API error`, network/DB errors) → falls through to 500

This means a Plaud 5xx after the client's retry budget surfaces as 500 with a generic "Verification failed" / "Failed to connect Plaud account" message instead of misleading the user into "fix your token" troubleshooting. Pinned by the new `isUserActionablePlaudError` test block in `src/tests/regressions/65-paste-token.test.ts` (5 cases covering 4xx, 5xx, bare-prefix, SSRF, unrelated).

File: src/lib/plaud/persist-connection.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            "Plaud API error: token validation failed (Plaud rejected the token on /device/list)",
+        );
+    }
+
+    // 3. Encrypt + upsert connection. Always re-scope by userId on the
+    // UPDATE/SELECT so we never touch another user's row (defense-in-depth
+    // alongside the userId-scoped lookup).
+    const encryptedAccessToken = encrypt(accessToken);
+
+    const [existingConnection] = await db
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/lib/plaud/persist-connection.ts]: Fixed in 7174477. Real race — `plaud_connections` has no unique constraint on `user_id` (kept additive to avoid breaking existing deployments that may have duplicates from this exact race), so two concurrent "Connect" clicks could each observe `existingConnection === undefined` and both insert.

`persistPlaudConnection` now wraps the connection upsert *and* device reconciliation in a single `db.transaction()`. The first statement inside is a per-user `pg_advisory_xact_lock(hashtextextended('plaud_connect:<userId>', 0))`, which serialises connect attempts for the same user. The lock releases automatically on commit/abort — no schema change, no migration risk on existing deployments.

Devices moved inside the same transaction so an aborted connect doesn't leave a half-written device list against the previous (rolled-back) token.

Postgres-only solution, which matches the project's existing DB constraint. Drizzle's `tx.execute(sql\`…\`)` is a stable API.

File: src/app/api/plaud/auth/connect-token/route.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        }
+
+        // Best-effort email enrichment. Failure is non-fatal \u2014
+        // plaud_connections.plaud_email is nullable.
+        const plaudEmail = await fetchPlaudUserMeEmail(accessToken, apiBaseRaw);
+
+        const source =
+            typeof body.source === "string" ? body.source : "unknown";
+        console.log(
+            `[plaud/connect-token] persisting connection (source=${source}, apiBase=${apiBaseRaw}, email=${plaudEmail ?? "unknown"})`,
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/app/api/plaud/auth/connect-token/route.ts]: Fixed in 7174477. Email dropped from the diagnostic log line in `src/app/api/plaud/auth/connect-token/route.ts`. `source` and `apiBase` are sufficient to disambiguate the connect path during debugging; the email isn't needed and shouldn't sit in server logs.

File: src/components/onboarding/onboarding-form.tsx [hunk 4]
```diff
@@ -320,27 +96,26 @@ export function OnboardingForm() {
...
+                            This is open source software \u2014 every line is
                             available for inspection:{" "}
                             <a
                                 href={GITHUB_REPO}
                                 target="_blank"
                                 rel="noopener noreferrer"
                                 className="underline decoration-dotted underline-offset-2 hover:text-muted-foreground/90 transition-colors"
                             >
-                                GitHub&nbsp;→
+                                GitHub&nbsp;\u2192
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/components/onboarding/onboarding-form.tsx]: Fixed in b225782. Replaced the literal `\u2192` in JSX text with the actual `→` character. Same root cause as the em-dash thread below — JSX text positions don't run JS unicode escapes the way string literals do. Swept the same fix across `plaud-connect-tabs.tsx` for consistency.

File: src/components/onboarding/onboarding-form.tsx [hunk 5]
```diff
@@ -320,27 +96,26 @@ export function OnboardingForm() {
...
+                            token is forwarded directly and never stored.
                         </p>
                         <p>
                             After login, your access token is encrypted with
                             AES-256-GCM and stored only on this self-hosted
                             instance. No data leaves your server.
                         </p>
                         <p>
-                            This is open source software — every line is
+                            This is open source software \u2014 every line is
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/components/onboarding/onboarding-form.tsx]: Fixed in b225782. Right call — JSX text doesn't interpret `\uXXXX` escapes (only JS string literals do), so the literal six-character sequence was rendering in the UI. Replaced with the actual `—` character. Same sweep covered `\u2192`, `\u2026`, `\u2190`, `\u00b7` across `plaud-connect-tabs.tsx` and `onboarding-form.tsx`.

File: src/lib/plaud/auth.ts [hunk 6]
```diff
@@ -115,3 +115,106 @@ export async function plaudVerifyOtp(
...
+ *
+ * Only HTTP 4xx (auth/permission/bad-input) is user-actionable. 5xx
+ * means Plaud is broken — surfacing those as 400 misleads users into
+ * "my token is bad" troubleshooting when in reality they should retry.
+ * Bare `Plaud API error` (no status) is treated as non-actionable too,
+ * since we can't tell which side of the wire the failure came from.
+ */
+export function isUserActionablePlaudError(message: string): boolean {
+    if (message === "Invalid API base") return true;
+    const m = /^Plaud API error \((\d{3})\):/.exec(message);
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/lib/plaud/auth.ts:146]: Good catch — fixed in 640f264.

Real regression in my previous P1 fix. Plaud's OTP and workspace helpers throw bare `Plaud API error: <msg>` (no parenthesised status) because Plaud returns HTTP 200 with a `status: -N` body for invalid verification codes, send-code rate limits, missing workspaces, etc. The narrowed regex returned false for those, so a user typing the wrong OTP would see a 500 "Verification failed" instead of 400 with Plaud's actual message.

`isUserActionablePlaudError` now treats bare-prefix messages as user-actionable while keeping the 5xx-vs-4xx distinction for HTTP-status-bearing ones:

- `Plaud API error: <msg>` (bare, business-level) → 400
- `Plaud API error (4xx): <msg>` → 400
- `Plaud API error (5xx): <msg>` → 500 (so users retry instead of being told "fix your token")
- `Invalid API base` → 400

The regression test block in `src/tests/regressions/65-paste-token.test.ts` flipped the bare-prefix assertion to `true` and added explicit cases for `invalid OTP`, `failed to send verification code`, and `no workspaces returned` so we don't regress this again.
